### PR TITLE
[4.0] Add filter to com_tags administration based on root tag

### PR DESCRIPTION
**Pull request by mattiaverga**
### Summary of Changes
This PR will add a new filter to the administration page of com_tags which makes possible to list only tags under a specific root tag.
This is useful when having a complex tag tree with a lot of levels: it lets you focus only on the desired root tag and its subtags.
When applied, the filter uses a recursive query to create a table with the subset of the root tag and its sub-tags, then the other filters are applied on this temporary table to get the final result.

### Testing Instructions
Create some main level tags and some nested tags under them.
Use the new filter tab to list only tags under one main tag (or under one of the nested tags).

### Expected result
When the filter is applied, the lists should show only the root tag and its child tags.

### Actual result
There is no such option to achieve that.

### Documentation Changes Required
The help section about com_tags should be updated with a paragraph for the new filter.

### Additional comments
There seems to be a bug in displaying the tag field: setting the multiple option to false prevents the hint to be showed:
immagine

### CODE REVIEW

File: administrator/language/en-GB/en-GB.com_tags.ini [hunk 1]
```diff
@@ -57,6 +57,8 @@ COM_TAGS_FIELDSET_OPTIONS="Options"
 COM_TAGS_FIELDSET_PUBLISHING="Publishing"
 COM_TAGS_FIELDSET_TAGGED_ITEMS="Tagged Items"
 COM_TAGS_FIELDSET_URLS_AND_IMAGES="Links and Images"
+COM_TAGS_FILTER_ROOT_DESC="List only tags under the specified root tag. The root tag is included in the search results."
+COM_TAGS_FILTER_ROOT_LABEL="Select root Tag"
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.com_tags.ini]:
```suggestion
COM_TAGS_FILTER_ROOT_LABEL="Select Root Tag"
```

**Comment by Eriiond** [comment]
I have tested this item :white_check_mark: successfully on 304219fcd1fa2bc9991e3033dc4caf2381756e4d

Selecting a parent tag also shows the children (as intended):
image

The only issue is that the proposal list does not contain the items:
image


<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24895">issues.joomla.org/tracker/joomla-cms/24895</a>.</sub>

**Comment by jduerscheid** [comment]
I have tested this item :red_circle: unsuccessfully on 304219fcd1fa2bc9991e3033dc4caf2381756e4d<br /><br />There is no items to select. You have to start typing the tag you want select. Futhermore the Label for the dropdown is missing.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24895">issues.joomla.org/tracker/joomla-cms/24895</a>.</sub>

**Comment by jduerscheid** [comment]
Screenshot to the failed test:screen shot 2019-10-19 at 13 30 29<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24895">issues.joomla.org/tracker/joomla-cms/24895</a>.</sub>

**Comment by mattiaverga** [comment]
> I have tested this item red_circle unsuccessfully on [304219f](https://github.com/joomla/joomla-cms/commit/304219fcd1fa2bc9991e3033dc4caf2381756e4d)
> 
> There is no items to select. You have to start typing the tag you want select. 

Yes, that's exactly how it works. It's the same for the tag filter in the Content component.


> Futhermore the Label for the dropdown is missing.

As I wrote in my first comment, that's a bug in how Joomla displays the tag filter. By setting the option `multiple="true"` the hint is displayed, but here we need `multiple="false"`, so it doesn't show the hint:
immagine

**Comment by SharkyKZ** [comment]
@mattiaverga can you please change "root" to "parent" everywhere? Root tag is an actual tag in `#__tags` table and its the parent of all other tags. So let's not confuse these two.

**Comment by mattiaverga** [comment]
> @mattiaverga can you please change "root" to "parent" everywhere? Root tag is an actual tag in `#__tags` table and its the parent of all other tags. So let's not confuse these two.

mmm I think that "parent" can be confusing: the filter will display all child levels under the chosen tag, not only the first level.
Is "base tag" more appropriate? Or maybe "starting"... I'm not an English native speaker, so I'm open to suggestions.

**Comment by roland-d** [comment]
@mattiaverga Do you have any interest in fixing the conflicts and picking this up or should the PR be closed?

**Comment by mattiaverga** [comment]
I have no more interest in this and currently I don't have free time to look at this, so I'll just close the PR

**Pull request closed without merging**